packages/api/src/command/medical/patient/update-patient.ts

```diff
@@ -22,42 +24,130 @@ export type PatientUpdateCmd = BaseUpdateCmdWithCustomer &
 // TODO build unit test to validate the patient is being sent correctly to Sequelize
 // See: document-query.test.ts, "send a modified object to Sequelize"
 // See: https://metriport.slack.com/archives/C04DMKE9DME/p1686779391180389
-export async function updatePatient(
-  patientUpdate: PatientUpdateCmd,
+export async function updatePatient({
+  patientUpdate,
   emit = true,
+  rerunPdOnNewDemographics = false,
+  augmentDemographics = false,
+  forceCommonwell,
+  forceCarequality,
+}: {
+  patientUpdate: PatientUpdateCmd;
+  emit?: boolean;
+  rerunPdOnNewDemographics?: boolean;
+  augmentDemographics?: boolean;
   // START TODO #1572 - remove
-  forceCommonwell?: boolean,
-  forceCarequality?: boolean
+  forceCommonwell?: boolean;
+  forceCarequality?: boolean;
   // END TODO #1572 - remove
-): Promise<Patient> {
+}): Promise<Patient> {
   const { cxId, facilityId } = patientUpdate;
 
   // validate facility exists and cx has access to it
-  const facility = await getFacilityOrFail({ cxId, id: facilityId });
-
-  const requestId = uuidv7();
-
-  const patientUpdateWithPD: PatientUpdateCmd = {
-    ...patientUpdate,
-    patientDiscovery: { requestId, startedAt: new Date() },
-  };
+  await getFacilityOrFail({ cxId, id: facilityId });
 
-  const patient = await updatePatientWithoutHIEs(patientUpdateWithPD, emit);
+  const patient = await updatePatientWithoutHIEs(patientUpdate, emit);
 
   const fhirPatient = toFHIR(patient);
   await upsertPatientToFHIRServer(patientUpdate.cxId, fhirPatient);
 
-  await cqCommands.patient.discover(patient, facility.id, requestId, forceCarequality);
-
-  await cwCommands.patient.update(
+  const updatedPatient = await scheduleOrRunPatientDiscovery({
     patient,
     facilityId,
-    getCqOrgIdsToDenyOnCw,
-    requestId,
-    forceCommonwell
-  );
+    rerunPdOnNewDemographics,
+    augmentDemographics,
+    forceCommonwell,
+    forceCarequality,
+  });
+
+  return updatedPatient;
+}
+
+// Needs to live somewhere else
+export async function scheduleOrRunPatientDiscovery({
```

Comment:
This should probably be split into 2 functions one for CW and one for CQ

Reply:
Or made to be reusable

Reply:
Yeah, I'd:
- move `scheduleOrRunPatientDiscovery` to `src/external/hie/command/` - this keeps files/logic that's not under `commonwell`, `carequality`, or `hie` to be "Metriport" specific, as in, independent of external integrations and easier to maintain
- split it into separate functions, each placed on the respective HIE (`src/external/commonwell/command/...`, same for CQ's)
- call them asynchronously

Reply:
In progress!

Reply:
This is done.